zeroconf/__init__.py

```diff
@@ -1073,44 +1082,96 @@ def write_record(self, record: DNSRecord, now: float) -> int:
         # Here is the short we adjusted for
         self.insert_short(index, length)
 
+        len_limit = _MAX_MSG_ABSOLUTE if allow_long else _MAX_MSG_TYPICAL
+
         # if we go over, then rollback and quit
-        if self.size > _MAX_MSG_ABSOLUTE:
+        if self.size > len_limit:
             while len(self.data) > start_data_length:
                 self.data.pop()
             self.size = start_size
-            self.state = self.State.finished
-            return 1
-        return 0
+            return False
+        return True
 
     def packet(self) -> bytes:
-        """Returns a string containing the packet's bytes
+        """Returns a string containing the first packet's bytes.
+
+        Generally, you want to use packets() in case the response
+        does not fit in a single packet, but this exists for
+        backward compatibility."""
+        p = self.packets()
```

Comment:
I think it's fine to just call the variable `packets`, not much saved by shortening it to just `p` and it can help avoiding some confusion.

Reply:
Ok, I changed it; I don't like local variable names shadowing class instances/functions, but I get that other people don't like short variable names.  My philosophy is that a variable names's length should be proportional to its scope and in this case the scope is tiny so I was comfortable with a single-letter variable name.